packages/core/src/timeline.ts

```diff
@@ -26,9 +26,22 @@ export class Timeline {
       this.plugins[type] = [plugin];
     }
     const settings = plugin.analytics?.settings.get();
-    if (settings) {
+    if (settings !== undefined) {
       plugin.update({ integrations: settings }, UpdateType.initial);
+    } else {
+      plugin.analytics?.settings.onChange((newSettings) => {
+        if (newSettings !== undefined) {
+          plugin.update({ integrations: newSettings }, UpdateType.initial);
+        }
+      });
     }
+
+    plugin.analytics?.settings.onChange((newSettings) => {
+      if (newSettings !== undefined) {
+        plugin.update({ integrations: newSettings }, UpdateType.refresh);
+      }
+    });
```

Comment:
These 2 subscriptions should be combined into one. Right now it will call update twice (one with initial type and the other with refresh type) since it doesn't unsubscribe the initial one.